### DISCUSSION TITLE
Outil d'essais unitaires et d'intégration

### DISCUSSION THREAD

**Discussion by houssamkhalifeh** [VilledeMontreal/MontrealEnCommun#8] [Q&A]
On cherche pour nos projets du programme un outil d'essais de code qui aura les caractéristiques suivantes:

- Disponible en logiciel libre
- Intégration DevOps avec des plateformes variées (GitHub, GitLab, BitBucket, Azure, AWS,..)
- Des essais préconfigurés permettant de commencer rapidement

J'ai une petite expérience avec SonarQube et j'ai aimé en particulier les tests de vulnérabilité (OWASP) et les rapports d'analyse. Solon regarde l'outil Squash,..

Avez-vous de l'expérience avec ces outils? Quel outil vous utilisez/recommandez sinon?
Merci!

**Comment by psionikangel** [comment]
La Ville de Montréal utilise en interne SonarQube pour tous ses projets et les résultats sont très bons en général. Facile à mettre en place et à utiliser.

**Comment by houssamkhalifeh** [comment]
@pagbokou-artm @MagaliCantine, @cliambrown Avez vous une recommandation?

**Reply by cliambrown** [reply]
Aucun de moi, désolé!

**Comment by Jduhommet** [comment]
Côté Solon, Squash nous intéresse aussi pour sa gestion des campagnes de tests et suivi des exigences fonctionnelles.Nous recherchons un outil de tests utilisateurs et fonctionnels, pas seulement pour intégration ou tests unitaires. De ce que j'ai vu SonarQube ne semble pas proposer cela ?
